tds/src/test/java/thredds/server/wfs/TestWFSController.java

```diff
@@ -0,0 +1,33 @@
+package thredds.server.wfs;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+
+import static org.junit.Assert.*;
+
+public class TestWFSController {
+
+    private WFSController wfs;
+    private MockHttpServletRequest request;
+    private MockHttpServletResponse response;
+    @Before
+    public void setUp(){
+        wfs = new WFSController();
+        request = new MockHttpServletRequest();
+        response = new MockHttpServletResponse();
+    }
+
+    @Test
+    public void describeFeatureTypeTest(){
+        request.addParameter("SERVICE", "WFS");
+        request.addParameter("REQUEST", "DescribeFeatureType");
+        request.addParameter("VERSION", "2.0.0");
+
+        wfs.httpHandler(request, response);

```

Comment:
This is a good start to the test! If you would like to build it out, you could consider what assertions you could make on the response. A few potentially relevant methods you could use to extract values for assertions:
https://docs.spring.io/spring/docs/4.2.4.RELEASE/javadoc-api/org/springframework/mock/web/MockHttpServletResponse.html#flushBuffer--
https://docs.spring.io/spring/docs/4.2.4.RELEASE/javadoc-api/org/springframework/mock/web/MockHttpServletResponse.html#getContentAsByteArray--
https://docs.spring.io/spring/docs/4.2.4.RELEASE/javadoc-api/org/springframework/mock/web/MockHttpServletResponse.html#getContentAsString--
https://docs.spring.io/spring/docs/4.2.4.RELEASE/javadoc-api/org/springframework/mock/web/MockHttpServletResponse.html#getContentLength--
https://docs.spring.io/spring/docs/4.2.4.RELEASE/javadoc-api/org/springframework/mock/web/MockHttpServletResponse.html#getStatus--

Tests on features related to WFS are not required prior to the demo. I like tests a lot, and sometimes find they speed up my development in the short term. In this case it is OK to charge the tech debt credit card if it is faster in the short term.